eval.go

```diff
@@ -40,14 +40,39 @@ func (m *FileMatch) addScore(what string, s float64) {
 }
 
 func (d *indexData) simplify(in query.Q) query.Q {
+	simpleShard := len(d.repoMetaData) == 1
 	eval := query.Map(in, func(q query.Q) query.Q {
 		switch r := q.(type) {
 		case *query.Repo:
-			return &query.Const{Value: strings.Contains(d.repoMetaData.Name, r.Pattern)}
+			for _, md := range d.repoMetaData {
+				if strings.Contains(md.Name, r.Pattern) {
+					if simpleShard {
+						return &query.Const{Value: true}
+					}
+					return q
+				}
+			}
+			return &query.Const{Value: false}
 		case *query.RepoBranches:
-			return r.Branches(d.repoMetaData.Name)
+			if simpleShard {
+				return r.Branches(d.repoMetaData[0].Name)
+			}
+			for _, md := range d.repoMetaData {
+				if _, ok := r.Set[md.Name]; ok {
+					return q
+				}
+			}
+			return &query.Const{Value: false}
 		case *query.RepoSet:
-			return &query.Const{Value: r.Set[d.repoMetaData.Name]}
+			for _, md := range d.repoMetaData {
+				if r.Set[md.Name] {
+					if simpleShard {
+						return &query.Const{Value: true}
+					}
```

Comment:
can we hoist the simpleShard check out of the for loop and keep the old conditional.

Reply:
done